Panic: "value is marked, so must be unmarked first" when entering random_string.suffix in terraform console

### Terraform Version

```shell
Terraform v1.15.5
on windows_amd64
+ provider registry.terraform.io/hashicorp/random v3.9.0
```

### Terraform Configuration Files

```terraform {
  required_providers {
    random = {
      source  = "hashicorp/random"
      version = "~> 3.9.0"
    }
  }
}

resource "random_string" "suffix" {
  length  = 6
  special = false
  upper   = false
}
```


### Debug Output

```
Not provided yet. The crash is reproducible with the minimal configuration above. I can provide TF_LOG=trace output if needed.
```


### Expected Behavior

Entering a resource reference in terraform console should display the resource object and its attributes, or return a normal Terraform error message if the reference is invalid.

Terraform should not terminate with a panic or crash.

### Actual Behavior

After running terraform apply and opening terraform console, entering:

random_string.suffix

causes Terraform to terminate unexpectedly with the following panic:

panic: value is marked, so must be unmarked first

Terraform exits/crashes instead of displaying the resource object or returning a standard error message.

Entering:

random_string.suffix.result

works as expected and returns the generated string value.

<details>
<summary>
STACK TRACE
</summary>

```
!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!

panic: value is marked, so must be unmarked first
goroutine 1 [running]:
runtime/debug.Stack()
        runtime/debug/stack.go:26 +0x5e
github.com/hashicorp/terraform/internal/logging.PanicHandler()
        github.com/hashicorp/terraform/internal/logging/panic.go:84 +0x18a
panic({0x7ff6c3a4d900?, 0x7ff6c4c009d0?})
        runtime/panic.go:783 +0x132
github.com/zclconf/go-cty/cty.Value.assertUnmarked(...)
        github.com/zclconf/go-cty@v1.18.0/cty/marks.go:191
github.com/zclconf/go-cty/cty.Value.True({{{0x7ff6c4c6b7f8?, 0x7ff6c4bf4528?}}, {0x7ff6c3ed9240?, 0xc000693e60?}})
        github.com/zclconf/go-cty@v1.18.0/cty/value_ops.go:425 +0x4d
github.com/hashicorp/terraform/internal/repl.FormatValue({{{0x7ff6c4c6b7f8, 0x7ff6c4bf4528}}, {0x7ff6c3ed9240, 0xc000693e60}}, 0x2)
        github.com/hashicorp/terraform/internal/repl/format.go:64 +0x492
github.com/hashicorp/terraform/internal/repl.formatMappingValue({{{0x7ff6c4c6bb10?, 0xc00072ba30?}}, {0x7ff6c3d66f60?, 0xc00048e3f0?}}, 0x0)
        github.com/hashicorp/terraform/internal/repl/format.go:148 +0x54f
github.com/hashicorp/terraform/internal/repl.FormatValue({{{0x7ff6c4c6bb10, 0xc00072ba30}}, {0x7ff6c3d66f60, 0xc00048e3f0}}, 0x0)
        github.com/hashicorp/terraform/internal/repl/format.go:71 +0x645
github.com/hashicorp/terraform/internal/repl.(*Session).handleEval(0xc000a71918, {0xc000a716d0?, 0x1?})
        github.com/hashicorp/terraform/internal/repl/session.go:86 +0x5a9
github.com/hashicorp/terraform/internal/repl.(*Session).Handle(0xc000a71918, {0xc000a716d0, 0x15})
        github.com/hashicorp/terraform/internal/repl/session.go:43 +0xbc
github.com/hashicorp/terraform/internal/command.(*ConsoleCommand).modeInteractive(0xc000506fc0, 0xc000c2d918, {0x7ff6c4c7de70, 0xc000740d80})
        github.com/hashicorp/terraform/internal/command/console_interactive.go:91 +0x52b
github.com/hashicorp/terraform/internal/command.(*ConsoleCommand).Run(0xc000506fc0, {0xc00010e3b0?, 0x0?, 0x0?})
        github.com/hashicorp/terraform/internal/command/console.go:174 +0xe98
github.com/hashicorp/cli.(*CLI).Run(0xc000913680)
```
</details>
        github.com/hashicorp/cli@v1.1.7/cli.go:265 +0x4ed
main.realMain()
        github.com/hashicorp/terraform/main.go:337 +0x1e12
main.main()
        github.com/hashicorp/terraform/main.go:62 +0x13

### Steps to Reproduce

1. `terraform init`

2. `terraform apply`

3. `terraform console`

4. `random_string.suffix`

### Additional Context

_No response_

### References

_No response_

### Generative AI / LLM assisted development?

_No response_

--- a/.changes/v1.15/BUG FIXES-20260603-172507.yaml
+++ b/.changes/v1.15/BUG FIXES-20260603-172507.yaml
@@ -1,0 +1,5 @@
+kind: BUG FIXES
+body: 'console: Fixed a panic caused by evaluating an expression involving deprecated values'
+time: 2026-06-03T17:25:07.111161+01:00
+custom:
+    Issue: "38676"

--- a/internal/repl/format.go
+++ b/internal/repl/format.go
@@ -20,11 +20,20 @@ func FormatValue(v cty.Value, indent int) string {
 	if !v.IsKnown() {
 		return "(known after apply)"
 	}
+
+	// Any marks on the value must either be used to return
+	// early or be removed. In future marks may used to cause
+	// values to be annotated, but the mark will still need to be
+	// removed before we can get a string representation of the value.
 	if marks.Has(v, marks.Sensitive) {
 		return "(sensitive value)"
 	}
 	if marks.Has(v, marks.Ephemeral) {
 		return "(ephemeral value)"
+	}
+	if marks.Has(v, marks.Deprecation) {
+		// Mark doesn't impact formatting, so remove to unblock normal formatting below
+		v, _ = v.Unmark()
 	}
 	if v.IsNull() {
 		ty := v.Type()

--- a/internal/repl/format_test.go
+++ b/internal/repl/format_test.go
@@ -181,6 +181,10 @@ EOT_`,
 			cty.StringVal("an ephemeral value").Mark(marks.Ephemeral),
 			"(ephemeral value)",
 		},
+		{
+			cty.StringVal("I'm a deprecated string value").Mark(marks.Deprecation),
+			`"I'm a deprecated string value"`, // We don't render deprecated values differently, but we need to ensure the deprecation mark doesn't interfere with formatting
+		},
 	}
 
 	for _, test := range tests {

